common/OpTestSystem.py

```diff
@@ -618,3 +624,132 @@ def sys_execute_cmd_onto_bmc(self, i_cmd):
             return self.cv_BMC.sys_execute_cmd_onto_bmc(i_cmd)
         except OpTestError as e:
             return BMC_CONST.FW_FAILED
+
+
+    ###########################################################################
+    # IPMI Console interfaces
+    ###########################################################################
+
+    ##
+    # @brief It will get the ipmi sol console
+    #
+    # @return l_con @type Object: it is a object of pexpect.spawn class or raise OpTestError
+    #
+    def sys_get_ipmi_console(self):
+        self.l_con = self.cv_IPMI.ipmi_get_console()
+        return self.l_con
+
+    ##
+    # @brief This function make sure, ipmi console is activated and then login to the lpar if host is already up.
+    #
+    # @param i_con @type Object: it is a object of pexpect.spawn class
+    #        and this is the console object used for ipmi sol console access and for lpar login.
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def sys_ipmi_lpar_login(self, i_con):
```

Comment:
OpTestSystem.py can't contain test login. It was designed to abstract function logic from other classes like OpTestLpar, OpTestBMC etc.

This function seems to belong to OpTestIPMI.py